modules/homa-lite/src/lib.rs

```diff
@@ -775,25 +843,51 @@ pub mod module {
 			Ok(())
 		}
 
+		/// Construct XCM message and sent it to the relaychain to withdraw_unbonded Staking
+		/// currency. The staking currency withdrew becomes available to be redeemed.
+		///
+		/// params:
+		/// 	- `staking_amount_unbonded`: amount of staking currency to withdraw unbond via XCM
 		#[transactional]
-		fn process_scheduled_unbond(staking_amount: Balance) -> DispatchResult {
-			let msg = Self::construct_xcm_unreserve_message(T::ParachainAccount::get(), staking_amount);
+		pub fn process_scheduled_unbond(staking_amount_unbonded: Balance) -> DispatchResult {
+			let msg = Self::construct_xcm_unreserve_message(T::ParachainAccount::get(), staking_amount_unbonded);
 
 			let res = pallet_xcm::Pallet::<T>::send_xcm(Here, Parent, msg);
 			log::debug!("on_idle XCM result: {:?}", res);
 			ensure!(res.is_ok(), Error::<T>::XcmFailed);
 
-			// Now that there's available staking balance, automatically match existing
-			// redeem_requests.
+			// Update storage with the new available amount
+			AvailableStakingBalance::<T>::mutate(|current| {
+				*current = current.saturating_add(staking_amount_unbonded);
+			});
+
+			Self::deposit_event(Event::<T>::ScheduledUnbondWithdrew(staking_amount_unbonded));
+			Ok(())
+		}
+
+		/// Iterate through all redeem requests, then match them with available_staking_balance.
+		/// This should be called when new available_staking_balance becomes available.
+		///
+		/// params:
+		/// 	- `max_num_matches`: Maximum number of redeem requests to be matched.
+		///
+		/// return:
+		/// 	Result<u32, DispatchError>: The number of redeem reqeusts actually matched.
+		#[transactional]
+		pub fn process_redeem_requests_with_available_staking_balance(
+			max_num_matches: u32,
+		) -> Result<u32, sp_runtime::DispatchError> {
```

Comment:
looks like this method doesn't fail? so should just return u32

Reply:
There's a few point that can return error. This function is doing the redeem work with available KSMs unbonded.